src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1196,6 +1196,45 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RemoveAllSrpSer
     return error;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_ClearSrpService(const char * aInstanceName, const char * aName)
+{
+    CHIP_ERROR error                         = CHIP_NO_ERROR;
+    typename SrpClient::Service * srpService = nullptr;
+
+    Impl()->LockThreadStack();
+
+    VerifyOrExit(aInstanceName, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(strlen(aInstanceName) <= SrpClient::kMaxInstanceNameSize, error = CHIP_ERROR_INVALID_STRING_LENGTH);
+    VerifyOrExit(aName, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(strlen(aName) <= SrpClient::kMaxNameSize, error = CHIP_ERROR_INVALID_STRING_LENGTH);
+
+    // Check if service to clear exists.
+    for (typename SrpClient::Service & service : mSrpClient.mServices)
+    {
+        if ((strcmp(service.mInstanceName, aInstanceName) == 0) && (strcmp(service.mName, aName) == 0))
+        {
+            srpService = &service;
+            break;
+        }
+    }
+
+    VerifyOrExit(srpService, error = MapOpenThreadError(OT_ERROR_NOT_FOUND));
+
+    ChipLogProgress(DeviceLayer, "clearing srp service: %s.%s", aInstanceName, aName);
+    error = MapOpenThreadError(otSrpClientClearService(mOTInst, &(srpService->mService)));
```

Comment:
Calling `otSrpClientClearService` without `otSrpClientAddService` releases the resources locally, but it doesn't release it on the SRP server which can be considered some kind of resource leak (as no one else can release the resources and we just forgot them ;)). I just can't think of a good reason for doing it.

Reply:
It's not resource leak, as it will be anyway removed by the server after lease interval expiration. And I can imagine that for some reason SRP server refuses removing service and we really want to do that now, so we do this locally. Nevertheless as discussed F2F there isn't any valid case for us to do this for now, so I'm going to remove ClearService API.